### PR TITLE
Driver version fix, Windows & macOS pipeline support

### LINKED ISSUE #138: Version checking doesn't work properly on mac or windows
There are commands for checking this in version_check.py that only work on linux, apparently...

### LINKED ISSUE #169: version_check on Windows: FileNotFoundError
I ran v0.12.0 on Windows 10, and got an exception that was not caught in version_check.py, line 28: FileNotFoundError.
I fixed it like this:
`    except (CalledProcessError, FileNotFoundError):`

All other Windows issues that I had in 0.11 seem to have been fixed 🥇

### PR DESCRIPTION
**Pull request by vonNiklasson**
## Summary

* Changed the approach to determine driver versions and existence.
  * Added a dynamic way to determine OS
  * Added different driver check implementations depending on OS
* Added macOS and Windows to GitHub Actions pipeline.
* Changed name of `logging` module to `logging_utils` to avoid confusion between the built in module with the same name.
* Deprecated the old `version_check` module.
* Added linting/disables for Windows check.

fixes #138 
fixes #169

### CODE REVIEW

File: webtraversallibrary/version_check.py [hunk 1]
```diff
@@ -8,18 +8,67 @@
...
 
-    CHROME = "google-chrome --version"
-    CHROMIUM = "chromium --version"
-    CHROMEDRIVER = "chromedriver --version"
-    FIREFOX = "firefox --version"
-    GECKODRIVER = "geckodriver --version"
+    @staticmethod
+    def deprecated_warning():
+        logger.warning(
+            "VERSION_CMD is deprecated and will be removed in future a version. "
```
Comment (redfungus) [inline webtraversallibrary/version_check.py:17]: Typo

```suggestion
            "VERSION_CMD is deprecated and will be removed in a future version."
```

File: webtraversallibrary/version_check.py [hunk 2]
```diff
@@ -8,18 +8,67 @@
...
+
+
+# Hack right now to preserve the attributes to the class but have them as callable properties
+VERSION_CMD = _VERSION_CMD()
 
 
 def run_cmd(cmd: str, title: str = None) -> bool:
     """Runs given command. Outputs data if descriptive title is given"""
+    logger.warning(
+        "version_check.run_cmd() is deprecated and will be removed in future a version. "
```
Comment (redfungus) [inline webtraversallibrary/version_check.py:69]:
```suggestion
        "version_check.run_cmd() is deprecated and will be removed in a future version."
```

File: tests/driver_check_test.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+
+def test_get_os_system_function_class():
+    assert isinstance(get_os_function_class(OS.LINUX), OsFunctionsLinux)
+    assert isinstance(get_os_function_class(OS.MACOS), OsFunctionsMacOs)
+    assert isinstance(get_os_function_class(OS.WINDOWS), OsFunctionsWindows)
+
+
+def test_function_class_get_is_driver_installed():
+    fc = get_os_function_class(OS.LINUX)
```
Comment (redfungus) [inline tests/driver_check_test.py:49]: What about mac and windows tests?

**Review by redfungus — APPROVED** [review]

File: webtraversallibrary/version_check.py (hunk 1 above)
Comment (vonNiklasson) [reply webtraversallibrary/version_check.py]: Thanks :)

File: tests/driver_check_test.py (hunk 3 above)
Comment (vonNiklasson) [reply tests/driver_check_test.py:49]: Good catch. I don't think I meant to have `OS.LINUX` there at all, making the function auto resolve the OS on its own. Will remove it and then they will all be covered by the platform check.

Comment (vonNiklasson) [reply tests/driver_check_test.py]: Fixed in 9b2ef8

**Pull request merged**